pkg/infra/httpclient/httpclientprovider/datasource_metrics_middleware.go

```diff
@@ -83,15 +84,20 @@ func DataSourceMetricsMiddleware() sdkhttpclient.Middleware {
 		}
 
 		labels := prometheus.Labels{"datasource": datasourceLabelName, "datasource_type": datasourceLabelType}
+		requestHistogramLabels := prometheus.Labels{
+			"datasource":                datasourceLabelName,
+			"datasource_type":           datasourceLabelType,
+			"secureSocksDSProxyEnabled": strconv.FormatBool(opts.ProxyOptions != nil && opts.ProxyOptions.Enabled),
+		}
 
-		return executeMiddlewareFunc(next, labels)
+		return executeMiddlewareFunc(next, labels, requestHistogramLabels)
 	})
 }
 
-func executeMiddleware(next http.RoundTripper, labels prometheus.Labels) http.RoundTripper {
+func executeMiddleware(next http.RoundTripper, labels prometheus.Labels, requestHistogramLabels prometheus.Labels) http.RoundTripper {
 	return sdkhttpclient.RoundTripperFunc(func(r *http.Request) (*http.Response, error) {
 		requestCounter := datasourceRequestCounter.MustCurryWith(labels)
-		requestHistogram := datasourceRequestHistogram.MustCurryWith(labels)
+		requestHistogram := datasourceRequestHistogram.MustCurryWith(requestHistogramLabels)
```

Comment:
Adding this label to all metrics would simplify - any reason not doing that?

Reply:
I decided not to do that at first because we won't use the others metrics. I'll do it to simplify then.